packages/react/cypress/component/auto/form/AutoForm.cy.tsx

```diff
@@ -47,6 +47,44 @@ describeForEachAutoAdapter("AutoForm", ({ name, adapter: { AutoForm }, wrapper }
     ensureFieldInputLabelsExist();
   });
 
+  it("onSuccess callback should return a record result after the form submission", () => {
+    const onSuccessSpy = cy.spy().as("onSuccessSpy");
+    cy.mountWithWrapper(<AutoForm action={api.widget.create} exclude={["gizmos"]} onSuccess={onSuccessSpy} />, wrapper);
+
+    ensureFieldInputLabelsExist();
+
+    cy.get(`input[name="widget.name"]`).type("test record");
+    cy.get(`input[name="widget.inventoryCount"]`).type("999");
+
+    cy.get(`input[id="Roles_Autocomplete_Textfield"]`).click().type("u"); // To search for "unauthenticated"
+    cy.get(`li[data-listbox-option-value="unauthenticated"]`).click();
```

Comment:
This could be a bit simpler if you limit the number of things that need to be selected in it -- i.e. just required fields, or name, or something similar

Reply:
I don't think we should really care much about roles here.